internal/pkg/v1/action/operator_update_test.go

```diff
@@ -0,0 +1,189 @@
+			WithScheme(sch).
+			Build()
+		cfg.Scheme = sch
+		cfg.Client = cl
+
+		return cfg
+	}
+
+	It("fails finding existing operator", func() {
+		cfg := setupEnv()
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "does-not-exist"
+		ext, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("not found"))
+		Expect(ext).To(BeNil())
+	})
+
+	It("fails to handle operator with non-catalog source type", func() {
+		cfg := setupEnv(buildExtension("test", withSourceType("unknown")))
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "test"
+		ext, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("unrecognized source type"))
+		Expect(ext).To(BeNil())
+	})
+
+	It("fails because desired operator state matches current", func() {
+		cfg := setupEnv(buildExtension(
+			"test",
+			withSourceType(olmv1.SourceTypeCatalog),
+			withConstraintPolicy(string(olmv1.UpgradeConstraintPolicyCatalogProvided))),
+		)
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "test"
+		ext, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err).To(MatchError(internalaction.ErrNoChange))
+		Expect(ext).To(BeNil())
+	})
+
+	It("fails because overridden desired operator state matches current", func() {
+		cfg := setupEnv(buildExtension(
+			"test",
+			withSourceType(olmv1.SourceTypeCatalog),
+			withConstraintPolicy(string(olmv1.UpgradeConstraintPolicyCatalogProvided)),
+			withChannels("a", "b"),
+			withLabels(map[string]string{"c": "d"}),
+			withVersion("10.0.4"),
+		))
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "test"
+		updater.OverrideUnset = true
+		ext, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err).To(MatchError(internalaction.ErrNoChange))
+		Expect(ext).To(BeNil())
+	})
+
+	It("fails validating operator version", func() {
+		cfg := setupEnv(buildExtension(
+			"test",
+			withSourceType(olmv1.SourceTypeCatalog),
+			withConstraintPolicy(string(olmv1.UpgradeConstraintPolicyCatalogProvided))),
+		)
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "test"
+		updater.Version = "10-4"
+		ext, err := updater.Run(context.TODO())
+
+		Expect(err).NotTo(BeNil())
+		Expect(err.Error()).To(ContainSubstring("parsing version"))
+		Expect(ext).To(BeNil())
+	})
+
+	It("fails updating operator", func() {
+		testExt := buildExtension(
+			"test",
+			withSourceType(olmv1.SourceTypeCatalog),
+			withConstraintPolicy(string(olmv1.UpgradeConstraintPolicyCatalogProvided)),
+		)
+		cfg := setupEnv(testExt)
+
+		ctx, cancel := context.WithCancel(context.TODO())
+		cancel()
+
+		updater := internalaction.NewOperatorUpdate(cfg.Client)
+		updater.Package = "test"
+		updater.Version = "10.0.4"
+		updater.Channels = []string{"a", "b"}
```

Comment:
Is the installed version from a or b? 
Is that a valid use case? same of: https://github.com/operator-framework/kubectl-operator/pull/225#discussion_r1985318900